src/components/pages/authProvider.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from 'react'
+import { Switch, Route, Router } from 'react-router'
+
+import { AuthProvider, ExpressValidationError, toCapitalizedWords } from 'devu-shared-modules'
+
+import config from 'config'
+
+import RequestService from 'services/request.service'
+import history from 'services/history.service'
+
+import LoadingOverlay from 'components/shared/loaders/loadingOverlay'
+import TextField from 'components/shared/inputs/textField'
+import ValidationErrorViewer from 'components/shared/errors/validationErrorViewer'
+
+import styles from './authProvider.scss'
+
+type ProviderSelectorProps = { providers: AuthProvider[]; onSelect: (provider: AuthProvider) => void }
+type ProivderFormProps = { provider?: AuthProvider }
+
+const ProviderSelector = ({ providers, onSelect }: ProviderSelectorProps) => (
+  <>
+    <h1 className={styles.header}>Select a Login Provider</h1>
+    <div className={styles.providerList}>
+      {providers.map((provider, index) => (
+        <button onClick={() => onSelect(provider)} key={index} className={styles.providerButton}>
+          Login with {provider.name}
+        </button>
+      ))}
+    </div>
+  </>
+)
+
+const ProviderForm = ({ provider }: ProivderFormProps) => {
+  if (!provider) {
+    history.push('/')
+    return null
+  }
+
+  const [formData, setFormData] = useState<Record<string, string>>({})
+  const [errors, setErrors] = useState<ExpressValidationError[] | Error>(new Array<ExpressValidationError>())
+
+  const { body: fields = [] } = provider
+
+  const handleUpdateForm = (key: string, value: string) => setFormData({ ...formData, [key]: value })
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault()
+
+    const { route } = provider
+
+    RequestService.post(`/api${route}`, formData, { credentials: 'include' })
+      .then(() => window.location.reload())
+      .catch(setErrors)
+  }
+
+  const hasErrors = (Array.isArray(errors) && errors.length > 0) || (!Array.isArray(errors) && errors.message)
+
+  return (
+    <>
+      <h1 className={styles.header}>{toCapitalizedWords(provider.name)}</h1>
```

Comment:
Since the name being used is being pulled from the yaml now directly, I'll just use the name raw 🥩. As for the description 😬  🤔  we _could_ use it. But I don't think it has a place on these buttons; it'd just be information overload. 

We are using it on auth that requires credentials directly on our page though (ie: dev auth)